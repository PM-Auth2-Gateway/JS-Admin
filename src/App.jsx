import React from 'react';
import { BrowserRouter as Router, Switch } from 'react-router-dom';

import styles from './App.module.scss';

import Navigation from './components/Navigation/Navigation';
import PrivateRoute from './components/PrivateRoute/PrivateRoute';
import ApplicationsPage from './pages/ApplicationsPage/ApplicationsPage';
import SingleApplicationPage from './pages/SingleApplicationPage/SingleApplicationPage';

function App() {
  return (
    <Router>
      <div className={styles.App}>
        <Navigation />
      </div>
      <div className={styles.container}>
        <Switch>
          <PrivateRoute exact path='/applications/:appId' component={SingleApplicationPage} />
          <PrivateRoute exact path='/applications' component={ApplicationsPage} />
        </Switch>
      </div>
    </Router>
  );
}

export default App;
