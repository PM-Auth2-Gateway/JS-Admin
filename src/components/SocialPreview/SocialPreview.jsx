import { Badge, ButtonGroup, Card } from 'react-bootstrap';

import styles from './SocialPreview.module.scss';
import SocialModal from '../SocialModal/SocialModal';
import DeleteSocialModal from '../DeleteSocialModal/DeleteSocialModal';

const SocialPreview = ({ id, name, isSetting }) => {
  return (
    <Card className={styles.card}>
      <Card.Body>
        <Card.Title className={styles.socialPreview}>
          <div className={styles.socialTitle}>
            <img className={styles.appIcon} alt='icon' />
            <p>{name}</p>
          </div>
          <Badge variant={isSetting ? 'success' : 'info'}>
            {isSetting ? 'Active' : 'Disabled'}
          </Badge>
        </Card.Title>
        <Card.Text className={'text-muted'}>
          Allow your users to login with their {name} Account
        </Card.Text>
        <ButtonGroup>
          <SocialModal id={id} mode={isSetting ? 'update' : 'create'} />
          {isSetting && <DeleteSocialModal id={id} />}
        </ButtonGroup>
      </Card.Body>
    </Card>
  );
};

export default SocialPreview;
