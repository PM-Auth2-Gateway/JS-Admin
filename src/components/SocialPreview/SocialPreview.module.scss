.card {
  margin-bottom: 20px;

  &:last-child {
    margin-bottom: 0;
  }

  a {
    text-decoration: none;
  }

  .socialPreview {
    display: flex;
    align-items: flex-start;
    justify-content: space-between;
    gap: 16px;

    .socialTitle {
      display: flex;
      align-items: flex-start;
      justify-content: space-between;

      img {
        margin-right: 8px;
      }
    }

    .appIcon {
      width: 32px;
      border-radius: 50%;
    }
  }
}
