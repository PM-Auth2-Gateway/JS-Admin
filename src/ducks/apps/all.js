import AppsApiService from '../../services/AppsApiService';

const LOAD_ALL = 'apps/all/load';
const ADD_ALL = 'apps/all/add';
const CLEAR_ALL = 'apps/all/clear';

const SET_LOADING = 'apps/all/loading/set';
const SET_ERROR = 'apps/all/error/set';

export const loadAllApps = () => async (dispatch) => {
  try {
    dispatch(setLoading(true));
    const res = await AppsApiService.get();

    dispatch(loadAll(res.data));
    dispatch(setError(null));
  } catch (error) {
    dispatch(setError(error));
  } finally {
    dispatch(setLoading(false));
  }
};

export const addApp = ({ name }) => async (dispatch) => {
  try {
    dispatch(setLoading(true));
    const res = await AppsApiService.post({ name });

    dispatch(addAll(res.data));
    dispatch(setError(null));
  } catch (error) {
    dispatch(setError(error));
  } finally {
    dispatch(setLoading(false));
  }
};

export const clearAllApps = () => (dispatch) => {
  dispatch(clearAll());
};

export const setAppError = (msg) => (dispatch) => {
  dispatch(setError(msg));
};

const loadAll = (apps) => ({
  type: LOAD_ALL,
  payload: apps,
});

const addAll = (app) => ({
  type: ADD_ALL,
  payload: app,
});

const clearAll = () => ({
  type: CLEAR_ALL,
});

const setLoading = (status) => ({
  type: SET_LOADING,
  payload: status,
});

const setError = (msg) => ({
  type: SET_ERROR,
  payload: msg,
});

const initialState = {
  value: [],
  loading: false,
  error: null,
};

export const reducer = (state = initialState, action = {}) => {
  switch (action.type) {
    case LOAD_ALL:
      return {
        ...state,
        value: [...state.value, ...action.payload],
      };
    case ADD_ALL:
      return {
        ...state,
        value: [...state.value, action.payload],
      };
    case CLEAR_ALL:
      return {
        ...state,
        value: [],
      };
    case SET_LOADING:
      return {
        ...state,
        loading: action.payload,
      };
    case SET_ERROR:
      return {
        ...state,
        error: action.payload,
      };
    default:
      return state;
  }
};
