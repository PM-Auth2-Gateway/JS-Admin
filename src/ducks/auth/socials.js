import LoginApiService from './../../services/LoginApiService';

const SET_LOADING = 'auth/socials/SET_LOADING';
const SET_SOCIALS = 'auth/socials/SET_SOCIALS';

const setLoading = (payload) => ({
  type: SET_LOADING,
  payload,
});

const setSocials = (payload) => ({
  type: SET_SOCIALS,
  payload,
});

export const getSocials = () => async (dispatch) => {
  dispatch(setLoading(true));
  const socials = await LoginApiService.getSocials();
  dispatch(setSocials(socials));
  dispatch(setLoading(false));
};

const initialState = {
  isLoading: false,
  socials: null,
  authLink: null,
};

const reducer = (state = initialState, { type, payload }) => {
  switch (type) {
    case SET_LOADING:
      return {
        ...state,
        isLoading: payload,
      };
    case SET_SOCIALS:
      return {
        ...state,
        socials: payload,
      };
    default:
      return state;
  }
};

export default reducer;
