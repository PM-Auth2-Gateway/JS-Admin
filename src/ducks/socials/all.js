import SocialsApiService from '../../services/SocialsApiService';
import { toast } from 'react-toastify';

const LOAD_ALL = 'socials/all/load';
const UPDATE_STATUS_BY_ID = 'socials/all/update_by_id/status';
const CLEAR_ALL = 'socials/all/clear';

const SET_LOADING = 'socials/all/loading/set';
const SET_ERROR = 'socials/all/error/set';

export const loadAllSocials = (appId) => async (dispatch) => {
  try {
    dispatch(setLoading(true));
    const res = await SocialsApiService.get(appId);

    dispatch(loadAll(res.data));
    dispatch(setError(null));
  } catch (error) {
    dispatch(setError(error));
  } finally {
    dispatch(setLoading(false));
  }
};

export const updateSocialStatusById = (appId, id, { status }) => async (
  dispatch
) => {
  try {
    dispatch(setLoading(true));
    const res = await SocialsApiService.updateStatusById(appId, id, { status });

    dispatch(
      updateStatusById(id, {
        status: res.data.is_active,
      })
    );
    dispatch(setError(null));

    toast(`Social network was ${!status ? 'disabled' : 'activated'}`);
  } catch (error) {
    dispatch(setError(error));

    toast.error('Something went wrong');
  } finally {
    dispatch(setLoading(false));
  }
};

export const clearAllSocials = () => (dispatch) => {
  dispatch(clearAll());
};

export const setSocialsError = (msg) => (dispatch) => {
  dispatch(setError(msg));
};

const loadAll = (apps) => ({
  type: LOAD_ALL,
  payload: apps,
});

const updateStatusById = (id, { status }) => ({
  type: UPDATE_STATUS_BY_ID,
  payload: {
    id,
    status,
  },
});

const clearAll = () => ({
  type: CLEAR_ALL,
});

const setLoading = (status) => ({
  type: SET_LOADING,
  payload: status,
});

const setError = (msg) => ({
  type: SET_ERROR,
  payload: msg,
});

const initialState = {
  value: [],
  loading: false,
  error: null,
};

export const reducer = (state = initialState, action = {}) => {
  switch (action.type) {
    case LOAD_ALL:
      return {
        ...state,
        value: action.payload,
      };
    case UPDATE_STATUS_BY_ID:
      return {
        ...state,
        value: state.value.map((elem) => ({
          ...elem,
          is_active:
            elem.id === action.payload.id
              ? action.payload.status
              : elem.is_active,
        })),
      };
    case CLEAR_ALL:
      return {
        ...state,
        value: [],
      };
    case SET_LOADING:
      return {
        ...state,
        loading: action.payload,
      };
    case SET_ERROR:
      return {
        ...state,
        error: action.payload,
      };
    default:
      return state;
  }
};
