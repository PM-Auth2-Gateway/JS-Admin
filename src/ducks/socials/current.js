import SocialsApiService from '../../services/SocialsApiService';

const SET_CURRENT = 'socials/current/set';
const UPDATE_CURRENT = 'socials/current/update';
const REMOVE_CURRENT = 'socials/current/remove';

const SET_LOADING = 'socials/current/loading/set';
const SET_ERROR = 'socials/current/error/set';

export const loadCurrentSocial = (appId, id) => async (dispatch) => {
  try {
    dispatch(setLoading(true));
    const res = await SocialsApiService.getById(appId, id);

    dispatch(setCurrent(res.data));
    dispatch(setError(null));
  } catch (error) {
    dispatch(setError(error));
  } finally {
    dispatch(setLoading(false));
  }
};

export const updateCurrentSocial = (
  appId,
  id,
  { client_id, scope, secret_key }
) => async (dispatch) => {
  try {
    dispatch(setLoading(true));
    const res = await SocialsApiService.updateById(appId, id, {
      client_id,
      scope,
      secret_key,
    });

    dispatch(updateCurrent(res.data));
    dispatch(setError(null));
  } catch (error) {
    dispatch(setError(error));
  } finally {
    dispatch(setLoading(false));
  }
};

export const removeCurrentSocial = () => (dispatch) => {
  dispatch(removeCurrent());
};

const setCurrent = (app) => ({
  type: SET_CURRENT,
  payload: app,
});

const updateCurrent = (data) => ({
  type: UPDATE_CURRENT,
  payload: data,
});

const removeCurrent = () => ({
  type: REMOVE_CURRENT,
});

const setLoading = (status) => ({
  type: SET_LOADING,
  payload: status,
});

const setError = (msg) => ({
  type: SET_ERROR,
  payload: msg,
});

const initialState = {
  value: {},
  loading: false,
  error: null,
};

export const reducer = (state = initialState, action = {}) => {
  switch (action.type) {
    case SET_CURRENT:
      return {
        ...state,
        value: action.payload,
      };
    case UPDATE_CURRENT:
      return {
        ...state,
        value: action.payload,
      };
    case REMOVE_CURRENT:
      return {
        ...state,
        value: {},
      };
    case SET_LOADING:
      return {
        ...state,
        loading: action.payload,
      };
    case SET_ERROR:
      return {
        ...state,
        error: action.payload,
      };
    default:
      return state;
  }
};
