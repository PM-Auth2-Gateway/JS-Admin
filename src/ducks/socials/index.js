import { combineReducers } from 'redux';

import { reducer as socialsAllReducer } from './all';
import { reducer as socialsCurrentReducer } from './current';

export const reducer = combineReducers({
  all: socialsAllReducer,
  current: socialsCurrentReducer,
});
