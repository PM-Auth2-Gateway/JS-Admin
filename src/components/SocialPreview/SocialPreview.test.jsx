import { shallow } from 'enzyme';
import { useDispatch } from 'react-redux';
import { useAppContext } from '../../contexts/App.context';

import { updateSocialStatusById } from '../../ducks/socials/all';

import SocialPreview from './SocialPreview';

jest.mock('react-redux', () => ({
  useDispatch: jest.fn(),
}));

jest.mock('../../contexts/App.context', () => ({
  useAppContext: jest.fn(),
}));

jest.mock('../../ducks/socials/all', () => ({
  updateSocialStatusById: jest.fn(),
}));

describe('SocialPreview Component', () => {
  const dispatch = jest.fn();
  const action = jest.fn();

  beforeAll(() => {
    updateSocialStatusById.mockReturnValue(action);
  });

  beforeEach(() => {
    useDispatch.mockReturnValue(dispatch);

    useAppContext.mockReturnValue({
      appId: 1,
    });
  });

  it('should render component', function () {
    const component = shallow(<SocialPreview />);

    expect(component).toMatchSnapshot();
  });
});
