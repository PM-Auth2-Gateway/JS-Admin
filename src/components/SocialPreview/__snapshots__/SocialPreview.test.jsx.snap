// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialPreview Component should render component 1`] = `
<Card
  body={false}
  className="card"
>
  <CardBody>
    <CardTitle
      className="socialPreview"
    >
      <div
        className="socialTitle"
      >
        <img
          alt="icon"
          className="appIcon"
        />
        <p />
      </div>
      <Badge
        pill={false}
        variant="secondary"
      >
        None
      </Badge>
    </CardTitle>
    <CardText
      className="text-muted"
    >
      Allow your users to login with their 
       Account
    </CardText>
    <ButtonGroup
      role="group"
      toggle={false}
      vertical={false}
    >
      <SocialModal
        mode="create"
      />
    </ButtonGroup>
  </CardBody>
</Card>
`;
